src/github.com/matrix-org/rageshake/main.go

```diff
@@ -69,17 +90,29 @@ func main() {
 	fs := http.StripPrefix("/api/listing/", ls)
 
 	// set auth if env vars exist
-	usr := os.Getenv("BUGS_USER")
-	pass := os.Getenv("BUGS_PASS")
+	usr := cfg.BugsUser
+	pass := cfg.BugsPass
 	if usr == "" || pass == "" {
-		fmt.Println("BUGS_USER and BUGS_PASS env vars not found. No authentication is running for /api/listing")
+		fmt.Println("No listings_auth_user/pass configured. No authentication is running for /api/listing")
 	} else {
 		fs = basicAuth(fs, usr, pass, "Riot bug reports")
 	}
 	http.Handle("/api/listing/", fs)
 
-	port := os.Args[1]
-	log.Println("Listening on port", port)
+	log.Println("Listening on", *bindAddr)
+
+	log.Fatal(http.ListenAndServe(*bindAddr, nil))
+}
 
-	log.Fatal(http.ListenAndServe(":"+port, nil))
+func loadConfig(configPath string) (*config, error) {
+	contents, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return nil, err
+	}
+	var cfg config
+	if err = yaml.Unmarshal(contents, &cfg); err != nil {
+		return nil, err
+	}
+	// check required fields
```

Comment:
Thanks. Maybe you should also fix the place I cargo-culted this from: https://github.com/matrix-org/dendrite/blob/master/src/github.com/matrix-org/dendrite/cmd/dendrite-sync-server/main.go#L47 :)